ValveResourceFormat/IO/GltfModelExporter.cs

```diff
@@ -503,24 +767,25 @@ private Material GenerateGLTFMaterialFromRenderMaterial(VMaterial renderMaterial
                     continue;
                 }
 
-                var bitmap = ((ResourceTypes.Texture)textureResource.DataBlock).GenerateBitmap();
+                string exportedTexturePath;
 
-                if (renderTexture.Key.StartsWith("g_tColor", StringComparison.Ordinal) && material.Alpha == AlphaMode.OPAQUE)
+                using (var bitmap = ((ResourceTypes.Texture)textureResource.DataBlock).GenerateBitmap())
                 {
-                    var bitmapSpan = bitmap.PeekPixels().GetPixelSpan<SKColor>();
-
-                    // expensive transparency workaround for color maps
-                    for (var i = 0; i < bitmapSpan.Length; i++)
+                    if (renderTexture.Key.StartsWith("g_tColor", StringComparison.Ordinal) && material.Alpha == AlphaMode.OPAQUE)
                     {
-                        bitmapSpan[i] = bitmapSpan[i].WithAlpha(255);
+                        var bitmapSpan = bitmap.PeekPixels().GetPixelSpan<SKColor>();
+
+                        // expensive transparency workaround for color maps
+                        for (var i = 0; i < bitmapSpan.Length; i++)
+                        {
+                            bitmapSpan[i] = bitmapSpan[i].WithAlpha(255);
+                        }
                     }
-                }
 
-                string exportedTexturePath = Path.Join(DstDir, fileName);
-                exportedTexturePath = Path.ChangeExtension(exportedTexturePath, "png");
+                    exportedTexturePath = Path.Join(DstDir, fileName);
```

Comment:
You can move these manipulations outside of the `using`.

Reply:
The last line in the using, bitmap.PeekPixels().Encode(), uses a FileStream which takes the exportedTexturePath in the constructor.

Reply:
Ah right, will fix.